model/auditconv.go

```diff
@@ -0,0 +1,663 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import "github.com/francoispqt/gojay"
+
+// AuditModelTypeConv converts key model types to something better suited for audit output.
+func AuditModelTypeConv(val interface{}) (newVal interface{}, converted bool) {
```

Comment:
I think this could use some unit tests

Reply:
Thanks @fm2munsh for your review.  I added unit tests for audit records converting meta fields as you pointed out below.  

As for the `AuditModelTypeConv` here, I can't think of any unit tests that add any real safety, but they do make code more brittle.  This is just a type switch that returns a data only struct for a bunch of model types.

Reply:
> As for the AuditModelTypeConv here, I can't think of any unit tests that add any real safety, but they do make code more brittle. This is just a type switch that returns a data only struct for a bunch of model types.

True, i guess I was just thinking a test that ensures that primitive types still return as expected and nil returns nil - not necessarily testing every single switch case

Reply:
Ok, I'll add that.  Thanks.

Reply:
Added unit test via:  https://github.com/mattermost/mattermost-server/pull/14144/commits/4d2a5945c7056023fb4c12bb87ecf3975870e940